### PR TITLE
add optional param for aws iam role name to allow assuming roles for cross-account access to k8s clusters

### PR DESCRIPTION
**Pull request by raids**
Haven't done much type/javascript, but I think this adds optional args to `setAWSCredentials` and appends to the `conf` variable (no longer a `const`) if it's passed in.

I'll `build` and push back up once it's approved.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -15,18 +15,31 @@ const shell = (cmd: string): string => {
...
   aws_access_key_id=${awsAccessKeyId}
   aws_secret_access_key=${awsSecretAccessKey}
   `
-  const conf = `
+  let conf = `
   [default]
   region=eu-west-1
   `
 
+  if (awsIamRoleName) {
```
Comment (meraymond2) [inline src/index.ts]: You could do something like this
```
const profile =
    awsIamRoleName && awsAccountId
      ? `
  [profile ${awsIamRoleName}]
  region=eu-west-1
  source_profile=default
  role_arn=arn:aws:iam::${awsAccountId}:role/${awsIamRoleName}
  `
      : ""
  const conf = `
  [default]
  region=eu-west-1

  ${profile}
  `
  ```
to get rid of the reassignment.

Comment (raids) [reply src/index.ts]: @meraymond2 oh, cool. I'd also thought of creating a separate function which would just append profiles to the config file which would make it easier to test and also maybe support adding multiple profiles if we should ever need that.

What do you think between your suggestion and splitting it out into a separate function?

Comment (meraymond2) [reply src/index.ts]: I think split it out, mine ends up looking weird after it's been run through prettier.

**Comment by raids** [comment]
@meraymond2 I split it out if you could take another look

**Review by meraymond2 — APPROVED** [review]

**Pull request merged**